### PR TITLE
Support for OpenAPI v3 data types

### LINKED ISSUE #8: OpenAPI3: Support Type Mappings
Currently, Truce uses a lot of Go specific primitive types.
OpenAPI3 has a number of its own primitives: https://swagger.io/specification/#data-types

My current suggestion is the following mappings:

truce  | openapi
------|---------
int | integer
string | string
bool | bool
float | number
[]byte | string (base64 encoded)
[]T | type: array - items: { type: T }
map[string]T | type: object - additionalProperties: { type: T }

### PR DESCRIPTION
**Pull request by GeorgeMac**
Closes #8 

Initial pass at support the open api data type mappings from the Truce types.

This also moves the structure around to reflect the following:
```
truce.go // still contains generated truce.cue into go
spec.go // still contains the in Go definition of the truce Specification object
internal/
    output/.gitkeep // will contain output code found in main.go
    generate/
        go/       # now contains Go code generators / writers for types, servers and clients
        openapi/  # now contains OpenAPI writer
```

### CODE REVIEW

**Review by brettbuddin — COMMENTED** [review]
Disregard this comment.

File: internal/cmd/trucegen/main.go [hunk 1]
```diff
@@ -31,5 +32,27 @@ func main() {
...
+
+		_, _ = fi.Write(line)
+	}
+
+	if err != io.EOF {
+		panic(err)
+	}
+
+	// write the final line
+	_, _ = fi.Write(line)
```
Comment (brettbuddin) [inline internal/cmd/trucegen/main.go:58]: Worth dropping a TODO with a link to your PR here.

File: internal/generate/openapi/openapi_test.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package openapi
+
+import (
+	"bytes"
+	"io/fs"
+	"os"
+	"testing"
+
+	"cuelang.org/go/cue"
+	"github.com/georgemac/truce"
```
Comment (brettbuddin) [inline internal/generate/openapi/openapi_test.go]: Looks like there are some stray imports from the old way.

**Review by brettbuddin — CHANGES_REQUESTED** [review]

File: internal/generate/openapi/openapi_test.go (hunk 2 above)
Comment (GeorgeMac) [reply internal/generate/openapi/openapi_test.go]: gross, good spot

**Review by brettbuddin — APPROVED** [review]

**Pull request merged**